### PR TITLE
JavaScript folder: update README.md

### LINKED ISSUE #24103: JavaScript folder: update README.md
**1. Description:**

READMEs need to be updated throughout the curriculum repository.

Related to: 
- https://github.com/TheOdinProject/top-meta/issues/109

### Course README template:

```markdown
# <course_name>

/* in case course belongs to multiple paths, make sure to link those paths */ 

This folder contains lesson markdown files that make up the <course_name> course. This course exists in the [<path_name>](<path_link>) path on the Odin Project Website.

## Course Outline

The following list represents how the lessons are divided into sections and presented on the website.

**Disclaimer:** Given the ever updating nature of the curriculum, the outline might be outdated. See the [<course_name> course on the website](<course_link>)
instead.

### <section_name>

/* make sure the lesson order corresponds to the order on the website */
/* make sure to use relative links with respect to this README to link the markdown file */

1. [<lesson_name>](<lesson_markdown_file_link>)
2. [**<project_name>**](<project_markdown_file_link>)
```

**2. Acceptance Criteria:**

- [ ] Update JavaScript folder's `README.md` file to the template provided above
- [ ] Remove comments from the template
- [ ] Fill in the appropriate information in the angle (`<>`) brackets in the template
  - Make sure to the path/course/lesson names correspond to what is on the website  
- [ ] Use relative links to link the markdown files
  - For example use `react_js/hooks.md` as the link to the markdown file instead of `https://github.com/TheOdinProject/curriculum/blob/main/javascript/react_js/hooks.md` 
  - Make sure the links work as intended
- [ ] Don't add the links to the lessons in the "Intermediate Git" section, since they are already covered in the [Git's folder README](https://github.com/TheOdinProject/curriculum/blob/main/git/README.md)

**3. Additional Information:**

If anyone wants to work on this, please comment to be assigned. In case of any questions, feel free to ask below.

### PR DESCRIPTION
**Pull request by nguyens20**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [ ] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [ ] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #24103 

**2. This PR:**

Update JavaScript folder's README.md file to the template provided above.
Remove comments from the template.
Check to make sure the path/course/lesson names correspond to what is on the website
Used relative links to link the markdown files

**3. Additional Information:**

### CODE REVIEW

File: javascript/README.md [hunk 1]
```diff
@@ -1,52 +1,60 @@
-Make your websites dynamic and interactive with JavaScript! You'll create features and stand-alone applications. This course will wrap everything you've learned at The Odin Project into one, final capstone project.
+# JavaScript
 
-## The Outline
+This folder contains lesson markdown files that make up the JavaScript course. This course exists in the [Full Stack JavaScript](https://www.theodinproject.com/paths/full-stack-javascript?/) path on the Odin Project Website.
```
Comment (01zulfi) [inline javascript/README.md:3]: JavaScript path exists in the Full Stack Ruby on Rails path as well:
```suggestion
This folder contains lesson markdown files that make up the JavaScript course. This course exists in the [Full Stack JavaScript](https://www.theodinproject.com/paths/full-stack-javascript?/) and the [Full Stack Ruby on Rails](add_link_here) paths on the Odin Project Website.
```

File: javascript/README.md [hunk 2]
```diff
@@ -1,52 +1,60 @@
...
+5. [Lifecycle Methods](react_js/lifecycle_methods.md)
+6. [Hooks](react_js/hooks.md)
+7. [**Project: Memory Card**](react_js/project_memory_card.md)
+8. [Router](react_js/router.md)
+9. [React Testing Part 1](react_js/react_testing_part_one.md)
+10. [React Testing Part 2](react_js/react_testing_part_two.md)
+11. [**Project: Shopping Cart**](react_js/project_shopping_cart.md)
+12. [Advanced Concepts](react_js/advanced_concepts.md)
+### JavaScript and the Backend
+1. [Using BaaS For Your Back End](javascript_and_the_backend/using_baas_for_your_backend.md)
```
Comment (01zulfi) [inline javascript/README.md:56]: The BaaS lesson is exclusive to the Full Stack JavaScript path, whereas we have a "Using Ruby on Rails For Your Backend" lesson exclusive to the Full Stack Ruby on Rails path. We should update the README to reflect this:
```suggestion
1. - [Using BaaS For Your Back End](javascript_and_the_backend/using_baas_for_your_backend.md) (Exclusive to the Full Stack JavaScript path)
   - [Using Ruby on Rails For Your Back End](add_file_link_here) (Exclusive to the Full Stack Ruby on Rails path)
```

File: javascript/README.md [hunk 3]
```diff
@@ -1,52 +1,60 @@
...
+2. [**Project: Where's Waldo (A Photo Tagging App**](javascript_and_the_backend/javascript_and_the_backend/project_wheres_waldo_a_photo_tagging_app.md)
```
Comment (01zulfi) [inline javascript/README.md:57]: The link for Where's Waldo project is incorrect. It needs to be updated.

**Review by 01zulfi — CHANGES_REQUESTED** [review]

File: javascript/README.md (hunk 3 above)
Comment (nguyens20) [reply javascript/README.md]: I have changed the project's link in the newest commit !

**Review by 01zulfi — APPROVED** [review]
Looks great! Thanks for working on this :fire:

**Pull request merged**